jax/_src/interpreters/batching.py

```diff
@@ -376,12 +380,9 @@ def process_call(self, call_primitive, f, tracers, params):
     sizes = (x.shape[d] if type(d) is int else len(d.segment_lengths)
              for x, d in zip(vals, dims) if d is not not_mapped)
     axis_size, = core.dedup_referents(sizes)
-    segment_lens, dims = unpack_concat_axes(dims)
     f_, dims_out = batch_subtrace(f, self.main, tuple(dims))
-    f_ = _update_annotation(f_, f.in_type, axis_size, self.axis_name, dims,
-                            segment_lens)
-    vals_out = call_primitive.bind(f_, *segment_lens, *vals, **params)
-    vals_out, dims_out = reassemble_concat_axes(vals_out, dims_out())
+    f_ = _update_annotation(f_, f.in_type, axis_size, self.axis_name, dims, [])
```

Comment:
This doesn't seem right in general; can you explain why we can always pass an empty list here, and skip the `unpack_concat_axis` / `reassemble_concat_axis` stuff? I assume we can't skip those things in general, but we don't have a test that's exercising this path. If so, we should figure out what needs to be done and leave a TODO (or else fix up this section).

Reply:
Let's work this one out synchronously.  We don't need to call a function literally called `unpack_concat_axis` because there are no more concat axes; whatever translation this was trying to do should be simpler in the padded-and-stacked representation.  That said, I'm not sure what translation this was trying to do, and what test would exercise it.  I passed an empty list because that's what `unpack_concat_axis` would do on a non-ragged array, which got the existing tests passing.

Reply:
In offline discussion, we determined: (i) this code seems to only relevant for getting axes to work under `jit`, which is why current tests pass without it, and (ii) we will revive an updated version in a future PR.